tests/test_docker_hostname_wrapper.py

```diff
@@ -0,0 +1,186 @@
+
+import mock
+import pytest
+
+from paasta_tools import docker_hostname_wrapper
+
+
+class TestParseEnvArgs(object):
+    def test_empty(self):
+        env = docker_hostname_wrapper.parse_env_args(['docker'])
+        assert env == {}
+
+    @pytest.mark.parametrize('args', [
+        ['docker', '-e', 'key=value'],
+        ['docker', '-e=key=value'],
+        ['docker', '-te', 'key=value'],  # -e can be combined with other short args
+        ['docker', '-et', 'key=value'],  # -t takes no additional parameters so docker allows this order
+    ])
+    def test_short(self, args):
+        env = docker_hostname_wrapper.parse_env_args(args)
+        assert env == {'key': 'value'}
+
+    @pytest.mark.parametrize('args', [
+        ['docker', '--env', 'key=value'],
+        ['docker', '--env=key=value'],
+    ])
+    def test_long(self, args):
+        env = docker_hostname_wrapper.parse_env_args(args)
+        assert env == {'key': 'value'}
+
+    @pytest.mark.parametrize('args', [
+        ['docker', '--eep', 'key=value'],  # ignore -e in double dashes unless its --env
+        ['docker', '-a', 'key=value'],  # only -e should matter
+        ['docker', '-'],  # just don't crash
+    ])
+    def test_short_invalid(self, args):
+        env = docker_hostname_wrapper.parse_env_args(args)
+        assert env == {}
+
+    def test_mixed_short_long(self):
+        env = docker_hostname_wrapper.parse_env_args(['docker', '-e', 'foo=bar', '--env=apple=banana', '--env', 'c=d'])
+        assert env == {'foo': 'bar', 'apple': 'banana', 'c': 'd'}
+
+    def test_multiple_equals(self):
+        env = docker_hostname_wrapper.parse_env_args(['docker', '-e', 'foo=bar=cat'])
+        assert env == {'foo': 'bar=cat'}
+
+    def test_dupe(self):
+        env = docker_hostname_wrapper.parse_env_args(['docker', '-e', 'foo=bar', '-e', 'foo=cat'])
+        assert env == {'foo': 'cat'}
+
+    def test_empty_value(self):
+        env = docker_hostname_wrapper.parse_env_args(['docker', '-e', 'foo=', '--env=bar='])
+        assert env == {'foo': '', 'bar': ''}
+
+
+class TestAlreadyHasHostname(object):
+    def test_empty(self):
+        assert docker_hostname_wrapper.already_has_hostname(['docker']) is False
+
+    @pytest.mark.parametrize('args', [
+        ['docker', '-h'],
+        ['docker', '-th'],
+        ['docker', '-ht'],
+    ])
+    def test_short(self, args):
+        assert docker_hostname_wrapper.already_has_hostname(args) is True
+
+    @pytest.mark.parametrize('args', [
+        ['docker', '-e=foo=hhh'],
+        ['docker', '--hhh'],
+        ['docker', '-'],
+    ])
+    def test_short_invalid(self, args):
+        assert docker_hostname_wrapper.already_has_hostname(args) is False
+
+    @pytest.mark.parametrize('args', [
+        ['docker', '--hostname', 'foo'],
+        ['docker', '--hostname=foo'],
+    ])
+    def test_long(self, args):
+        assert docker_hostname_wrapper.already_has_hostname(args) is True
+
+
+class TestGenerateHostname(object):
```

Comment:
Can you add some more tests with realistic mesos ids? I'm concerned about parsing this "user" input.

Here are two different classes:
```
ct:1487804100000:0:thirdparty_feeds thirdparty_feeds-cloudflare-all:
paasta--canary.main.git332d4a22.config458863b1.0126a188-f944-11e6-bdfb-12abac3adf8c
```

rpartition those! :)